tests/integration/test_ingress_relation.py

```diff
@@ -55,13 +60,18 @@ def start_server(self):
         deploy_any_charm(json.dumps(src_overwrite)),
         build_and_deploy_ingress(),
     )
+
     await ingress.set_config({"service-hostname": "any"})
     await model.wait_for_idle()
-    await run_action("any", "rpc", method="start_server")
+    await asyncio.sleep(5)
```

Comment:
If you need these `sleep` statements, there's probably something wrong with the charm. It should report ready status only when operational

Reply:
there seems to be some strange race condition at the moment, where the url only appears sometimes. the original integration test was broken initially, except it never checked the relation so it went un-noticed

Reply:
We discussed this IRL and the consensus was to update anycharm to go into Blocked status until it receives the info it needs over the relation. We should then be able to rely on `wait_for_idle`.